File: backend/src/main/java/com/plugga/backend/service/DeckCardServiceImpl.java
package com.plugga.backend.service;

import com.plugga.backend.dao.DeckCardDAO;
import com.plugga.backend.entity.DeckCard;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
public class DeckCardServiceImpl implements DeckCardService {
    private DeckCardDAO deckCardDAO;

    @Autowired
    public DeckCardServiceImpl(final DeckCardDAO deckCardDAO) {
        this.deckCardDAO = deckCardDAO;
    }

    @Override
    @Transactional
    public List<DeckCard> findAll() {
        return deckCardDAO.findAll();
    }

    @Override
    @Transactional
    public DeckCard findById(final int id) {
        return deckCardDAO.findById(id);
    }

    @Override
    @Transactional
    public List<DeckCard> findByDeckId(final int id) {
        return deckCardDAO.findByDeckId(id);
    }

    @Override
    @Transactional
    public List<DeckCard> findByCardId(final int id) {
        return deckCardDAO.findByCardId(id);
    }

    @Override
    @Transactional
    public void save(final DeckCard deckCard) {
        deckCardDAO.save(deckCard);
    }

    @Override
    @Transactional
    public void deleteById(final int id) {
        deckCardDAO.deleteById(id);
    }
}

File: backend/src/main/kotlin/com/plugga/backend/service/DeckCardServiceImpl.kt
package com.plugga.backend.service

import com.plugga.backend.repository.DeckCardRepository
import com.plugga.backend.entity.DeckCard
import java.util.Optional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class DeckCardServiceImpl @Autowired
constructor(private val deckCardRepository: DeckCardRepository) : DeckCardService {

    @Transactional
    override fun findAll(pageable: Pageable): Page<DeckCard> {
        return deckCardRepository.findAll(pageable)
    }

    @Transactional
    override fun findById(id: Int): DeckCard? {
        val queryResult: Optional<DeckCard> = deckCardRepository.findById(id)
        return if (queryResult.isPresent) queryResult.get() else null
    }

    @Transactional
    override fun findByDeckId(pageable: Pageable, id: Int): Page<DeckCard> {
        return deckCardRepository.findByDeckId(pageable, id)
    }

    @Transactional
    override fun findByCardId(pageable: Pageable, id: Int): Page<DeckCard> {
        return deckCardRepository.findByCardId(pageable, id)
    }

    @Transactional
    override fun save(deckCard: DeckCard) {
        deckCardRepository.save(deckCard)
    }

    @Transactional
    override fun deleteById(id: Int) {
        deckCardRepository.deleteById(id)
    }
}
